java/013-strstr.md

```diff
@@ -4,9 +4,32 @@
 - [next: 014. First Position of Target](014-first-position-of-target.md)
 
 ---
-
-put your own notes and solutions here.
-you can add any reference link such as [title](reference url) here.
+###Note:
+[substring](http://www.tutorialspoint.com/java/java_string_substring.htm)
+###Solution:
+```java
+class Solution {
+    /**
+     * Returns a index to the first occurrence of target in source,
+     * or -1  if target is not part of source.
+     * @param source string to be scanned.
+     * @param target string containing the sequence of characters to match.
+     */
+    public int strStr(String source, String target) {
+        if(source == null || target == null || target.length()>source.length()) 
+            return -1;
+        for(int i=0;i<source.length()-target.length()+1;i++){
+            // substring(beg_ind, end_ind), includes beg_ind, excludes end_ind
+            String substr = source.substring(i,i+target.length());
```

Comment:
You are right, I just fixed this. In the beginning I used this approach to compare chars one by one, but not managed to output the right index, so I tried the alternative of substring thing. You are right the substring would require more space so that it is not efficient, I have modified my repo and now it loos neat and great!